boards/nucleo-l496zg/include/periph_conf.h

```diff
@@ -156,6 +156,50 @@ static const spi_conf_t spi_config[] = {
 #define SPI_NUMOF           ARRAY_SIZE(spi_config)
 /** @} */
 
+/**
+ * @brief ADC configuration
+ *
+ * Note that we do not configure all ADC channels,
+ * and not in the STM32L496ZG order.  Instead, we
+ * just define 6 ADC channels, for the Nucleo
+ * Arduino header pins A0-A5 and the internal VBAT channel.
+ *
+ * To find appropriate device and channel find in the
+ * board manual, table showing pin assignments and
+ * information about ADC - a text similar to ADC[X]_IN[Y],
+ * where:
+ * [X] - describes used device - indexed from 0,
+ * for example ADC1_IN10 is device 0,
+ * [Y] - describes used channel - indexed from 1,
+ * for example ADC1_IN10 is channel 10
+ *
+ * For Nucleo-L496ZG this information is in board manual,
+ * Table 11, page 38.
+ *
+ * @{
+ */
+static const adc_conf_t adc_config[] = {
+    { .pin = GPIO_PIN(PORT_A, 3), .dev = 0, .chan =  8 }, /* ADC12_IN8   */
+    { .pin = GPIO_PIN(PORT_C, 0), .dev = 0, .chan =  1 }, /* ADC123_IN1  */
+    { .pin = GPIO_PIN(PORT_C, 3), .dev = 0, .chan =  4 }, /* ADC123_IN4  */
+    { .pin = GPIO_PIN(PORT_C, 1), .dev = 0, .chan =  2 }, /* ADC123_IN2  */
+    { .pin = GPIO_PIN(PORT_C, 4), .dev = 0, .chan = 13 }, /* ADC12_IN13  */
+    { .pin = GPIO_PIN(PORT_C, 5), .dev = 0, .chan = 14 }, /* ADC12_IN14  */
+    { .pin = GPIO_UNDEF, .dev = 0, .chan = 18 }, /* VBAT */
```

Comment:
```suggestion
    /* VBAT is connected ADC1_IN18 or ADC3_IN18 and a voltage divider
     * is used, so that only 1/3 of the actual VBAT is measured. This
     * allows for a supply voltage higher than the reference voltage. */
    { .pin = GPIO_UNDEF, .dev = 0, .chan = 18 },
```

Maybe you could also add the keyword for searching the datasheet here as done for the other ADC lines?

Reply:
Thanks for adding this information - before I only know about ADC1_IN18. 
I found more details in MCU datasheet and add it to your comment.